src/ts/TableSorting.class.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @author Marcel Bolten <github@marcelbolten.de>
+ * @copyright 2012 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+
+export default class TableSorting {
+  table: HTMLTableElement;
+  prevSortIcon: HTMLElement;
+  usesThead: boolean;
+
+  constructor(table: HTMLTableElement, usesThead = false) {
+    this.usesThead = usesThead;
+    this.makeSortable(table);
```

Comment:
I still need to find a way to add the `data-table-sort=true` attribute to tables created by tinymce.
Maybe you have an idea?
For now there is a workaround without using any id attribute in a way that would generate an error/crash.

Reply:
Yeah, didn't think about that. What I suggest then is that you can either init() and it will look for all the data-table-sort, OR you can init('tableID') and it will only work on that table id (and maybe on other data-table-sort present). Or init('auto') and it will do its stuff on any table it can find.

Reply:
I found a way to add a toggle button to the tinyMCE toolbar that will (un)set the 'data-table-sort' attribute.
The button has some features:
- It is only active (clickable) when a table is selected
- It will be highlighted (like e.g. the bold button) when 'data-table-sort' attribute is set
- It will add the action to the undoManager
- An alert is show if table cannot be sortable because of rowspan/colspan

The icon of the button could be changed to something more table-like.

What the button so far does not do is automatically set the top row cells from `td` to `th`. While this could obviously be done I don't think it is a good idea as this might modify tables in unwanted ways. But this is up for discussion.

Reply:
Just keep it simple for now. We can see later if it can be improved.